src/rom_workflows.cc

```diff
@@ -721,14 +782,142 @@ void computeRhoOnSamplePts(const CAROM::Matrix &dm,
     // rescaleTotalCharge();
 }
 
+template <class OrbitalsType>
+void testROMIonDensity(MGmolInterface *mgmol_)
+{
+    /* random number generator */
+    static std::random_device rd;  // Will be used to obtain a seed for the random number engine
+    static std::mt19937 gen(rd()); // Standard mersenne_twister_engine seeded with rd(){}
+    static std::uniform_real_distribution<> dis(0.0, 1.0);
+
+    Control& ct              = *(Control::instance());
+    Mesh* mymesh             = Mesh::instance();
+    const int subdivx = mymesh->subdivx();
+    const pb::PEenv& myPEenv = mymesh->peenv();
+    MGmol_MPI& mmpi      = *(MGmol_MPI::instance());
+    const int rank = mmpi.mypeGlobal();
+    const int nprocs = mmpi.size();
+
+    ROMPrivateOptions rom_options = ct.getROMOptions();
+
+    /* Load MGmol pointer and Potentials */
+    MGmol<OrbitalsType> *mgmol = static_cast<MGmol<OrbitalsType> *>(mgmol_);
+    Poisson *poisson = mgmol->electrostat_->getPoissonSolver(); 
+    Potentials& pot = mgmol->getHamiltonian()->potential();
+    const int dim = pot.size();
+    std::shared_ptr<Ions> ions = mgmol->getIons();
+
+    assert(!(mgmol->electrostat_->isDielectric()));
+    assert(pot.getBackgroundCharge() <= 0.0);
+
+    const int num_snap = 3;
+
+    /* 3 fictitious ion configurations */
+    const std::vector<Ion*> &overlappingVL_ions(ions->overlappingNL_ions());
```

Comment:
This is not a list of ions local to an MPI task. If it is meant to work in a distributed context, we need to do it differently.

Reply:
When I see `Potentials::initialize` (only function that uses `Potentials::initializeRadialDataOnMesh` to evaluate `rho_comp_`), it is looping over `overlappingVL_ions`. So in this case is it looping over the global list of ions?

Reply:
Ions are now globally set using `Ions::setPositions`. See Issue #284 .